submissions/asaMitaka/frogger/app.js

```diff
@@ -0,0 +1,169 @@
+// Enemies our player must avoid
+var newP = document.createElement('p');
+newP.textContent = `Your score: 0 `;
+document.body.append(newP);
+
+var enemyStat = {
+    width: 98,
+    height: 56,
+    sprite: "images/enemy-bug.png",
+};
+
+var field = {
+    min: -50,
+    borderX: 400,
+    borderY: 450,
+    waterCoord: 50,
+}
+
+var playerStat = {
+    width: 80,
+    height: 80,
+    startPositionX: 200,
+    startPositionY: 400
+}
+
+var xColDistance = 80;
+var yColDistance = 60;
+
+var Enemy = function(posX, posY, speed) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    this.posX = posX;
+    this.posY = posY;
+    this.speed = speed;
+    this.width = enemyStat.width;
+    this.height = enemyStat.height;
+    this.sprite = enemyStat.sprite;
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    this.posX += this.speed * dt;
+    if (this.posX > ctx.canvas.width) {
+        this.posX = -enemyStat.width;
+    }
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    this.collision();
+};
+
+Enemy.prototype.collision = function() {
+    if (player.posX < this.posX + xColDistance && player.posX + xColDistance > this.posX && player.posY < this.posY + yColDistance && yColDistance + player.posY > this.posY) {
+        player.goToStartPosition();
+    }
+}
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.posX, this.posY);
+};
+
+var allEnemyVal = [
+    {
+        x: -120,
+        y: 200,
+        speed: 60
+    },
+    {
+        x: -120,
+        y: 120,
+        speed: 120
+    },
+    {
+        x: -30,
+        y: 50,
+        speed: 300
+    }
+];
+
+var allEnemies = allEnemyVal.map(
+    ({x, y, speed}) => new Enemy(x, y, speed)
+)
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+var Player = function(posX, posY) {
+    this.posX = posX;
+    this.posY = posY;
+    this.stat = 0;
+    this.step = 75;
+
+    this.width = playerStat.width;
+    this.height = playerStat.height;
+
+    this.sprite = 'images/char-boy.png';
+}
+
+
+Player.prototype.update = function() {
+    this.win();
+}
+
+Player.prototype.win = function() {
+    if (field.min >= this.posY) {
+        console.log(field.min, this.posY);
```

Comment:
`console.log` is fine for debugging but should be removed from the final version of the code.